Automattic/woocommerce-subscriptions-core
# Define build tool version requirements for consistent dev/build environments

Fixes #209 

## Description


This PR:
* Adds a **Prerequisites** section to the repo’s  `README.md` which lists pre-requisite tools and the intention to use LTS versions.
	* Node: LTS (currently v16.17.1)
	* Composer: LTS (currently v2.2.18)
	* [nvm](https://github.com/nvm-sh/nvm#nvmrc) (optional and recommended for using the required node version)
* Defines the required **Node** version in:
   * `.nvmrc` (`lts/*`)
   * [`package.json/engines`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines) (node v16.17.1 npm v8.15.0)





## How to test this PR





1. Install [nvm](https://github.com/nvm-sh/nvm#nvmrc) if you haven't already
2. Run `nvm install`. The current LTS version of node/npm should be installed (node v16.17.1 npm v8.15.0)
3. Run `npm install`. No errors should occur.
4. [Download and install the LTS version of Composer](https://getcomposer.org/download/) as defined in README.md on this branch.
5. Run `composer install`
6. Run `npm run build`. The build should occur with no errors.
7. Run `npm run build` a second time.
8. Examine the contents of `woocommerce-subscriptions-core.zip`. 
   - `node_modules` should not be included.
   - `woocommerce-subscriptions-core.zip` should not be included.
9. Upload & install `woocommerce-subscriptions-core.zip` to a test WP site (e.g. JN) alongside WC Subs or WCPay and ensure that plugin can be activated without errors.
 
## Product impact
